Bugfix/unr 4478 windows dll crash fix (#2799)
spatialos/UnrealGDK: CHANGELOG.md

- Fixed a rare crash where a RepNotify callback can modify a GDK data structure being iterated upon.
- Fixed race condition in Spatial Test framework that would cause tests to time out with one or more workers not ready to begin the test.
- Fixed a assertion being triggered on async loaded entities due to queuing some component addition

> - Fix a crash opening session frontend with VS 16.8.0 and a bundled dbhelp.dll.

Reviewer: ```suggestion
- Fixed a crash that occured when opening the session frontend with VS 16.8.0 using a bundled dbghelp.dll.
```
